pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ScalarFunctions.java

```diff
@@ -56,6 +56,14 @@ public static byte[] stPoint(double x, double y) {
    * @return the created point
    */
   @ScalarFunction
+  public static byte[] stPoint(double x, double y, int isGeography) {
+    return stPoint(x, y, isGeography == 1);
+  }
+
+  /**
+   * TODO: Add @ScalarFunction annotation when we can support the same function name, number of parameters with a
+   *       different type.
+   */
   public static byte[] stPoint(double x, double y, boolean isGeography) {
```

Comment:
let's wait until #11726 
once merged we can change the 3rd arg to be Object and it will utilize the Transform function signature to match this one and we can convert based on what 3rd argument is by checking the `isInstanceOf`

Reply:
#11726 is merged, we should change the boolean to Object and check the 3rd argument whether it is a boolean or integer, simultaneously we also need to change TransformFunctionType ST_POINT to allow 3rd argument to be either Numeric or Boolean

Reply:
done